File: MSRtailHill/MSRtailHill.m

function MSRtailHill
clc
close all
a      = load('Bay9906_close_2kPoints.txt','-ascii');
b      = load('Bmw9906_close_2kPoints.txt','-ascii');
c      = load('Sie9906_close_2kPoints.txt','-ascii');
d      = a + b + c;
x      = log(d(1:end-1)./d(2:end)); %negative log-returns
n      = length(x);
x      = sort(x,'descend');
k      = 100;
x      = x(1:k,:);
gammaH = mean(log(x))-log(x(k))
alphaH = (mean(log(x))-log(x(k)) )^(-1)
sigmaH = x(k)*(k/n)^(1/alphaH)

File: MSRtailHill/MSRtailHill.R
rm(list=ls(all=TRUE))
#setwd("C:/...")

a  = read.table('Bay9906_close_2kPoints.txt')
b  = read.table('Bmw9906_close_2kPoints.txt')
c  = read.table('Sie9906_close_2kPoints.txt')
d  = a + b + c

n1 = dim(d)[1]
x  = log(d[1:n1-1, ]/d[2:n1,]) #negative log-returns
n  = length(x)
x  = sort(x, decreasing = TRUE)
k  = 100
x  = x[1:k]

gammaH = mean(log(x))-log(x[k])
alphaH = (mean(log(x)) -log(x[k]))^(-1)
sigmaH = x[k]*(k/n)^(1/alphaH)
sigmaH